R/derive_merged.R

```diff
@@ -173,136 +180,188 @@
-#'   "PILOT01",    "VS", "01-1302",  "WEIGHT",    "WEEK 2",     81.19,      "kg", "2013-09-15",
-#'   "PILOT01",    "VS", "01-1302",  "WEIGHT",    "WEEK 4",     82.56,      "kg", "2013-09-24",
-#'   "PILOT01",    "VS", "01-1302",  "WEIGHT",    "WEEK 6",     80.74,      "kg", "2013-10-08",
-#'   "PILOT01",    "VS", "01-1302",  "WEIGHT",    "WEEK 8",      82.1,      "kg", "2013-10-22",
-#'   "PILOT01",    "VS", "01-1302",  "WEIGHT",   "WEEK 12",      82.1,      "kg", "2013-11-05",
-#'   "PILOT01",    "VS", "17-1344",  "HEIGHT", "SCREENING",     163.5,      "cm", "2014-01-01",
-#'   "PILOT01",    "VS", "17-1344",  "WEIGHT", "SCREENING",     58.06,      "kg", "2014-01-01",
-#'   "PILOT01",    "VS", "17-1344",  "WEIGHT",  "BASELINE",     58.06,      "kg", "2014-01-11",
-#'   "PILOT01",    "VS", "17-1344",  "WEIGHT",    "WEEK 2",     58.97,      "kg", "2014-01-24",
-#'   "PILOT01",    "VS", "17-1344",  "WEIGHT",    "WEEK 4",     57.97,      "kg", "2014-02-07",
-#'   "PILOT01",    "VS", "17-1344",  "WEIGHT",    "WEEK 6",     58.97,      "kg", "2014-02-19",
-#'   "PILOT01",    "VS", "17-1344",  "WEIGHT",    "WEEK 8",     57.79,      "kg", "2014-03-14"
-#' )
+#'   ~DOMAIN,  ~USUBJID, ~VSTESTCD, ~VISIT,      ~VSSTRESN, ~VSDTC,
+#'   "VS",     "01",     "HEIGHT",  "SCREENING",     178.0, "2013-08-20",
+#'   "VS",     "01",     "WEIGHT",  "SCREENING",      81.9, "2013-08-20",
+#'   "VS",     "01",     "WEIGHT",  "BASELINE",       82.1, "2013-08-29",
+#'   "VS",     "01",     "WEIGHT",  "WEEK 2",         81.9, "2013-09-15",
+#'   "VS",     "01",     "WEIGHT",  "WEEK 4",         82.6, "2013-09-24",
+#'   "VS",     "02",     "WEIGHT",  "BASELINE",       58.6, "2014-01-11"
+#' ) %>%
+#'   mutate(STUDYID = "AB42")
 #'
 #' dm <- tribble(
-#'   ~STUDYID,  ~DOMAIN,  ~USUBJID, ~AGE,   ~AGEU,
-#'   "PILOT01",    "DM", "01-1302",   61, "YEARS",
-#'   "PILOT01",    "DM", "17-1344",   64, "YEARS"
-#' )
-#'
+#'   ~DOMAIN, ~USUBJID, ~AGE, ~AGEU,
+#'   "DM",    "01",       61, "YEARS",
+#'   "DM",    "02",       64, "YEARS",
+#'   "DM",    "03",       85, "YEARS"
+#' ) %>%
+#'   mutate(STUDYID = "AB42")
 #'
-#' # Merging all dm variables to vs
 #' derive_vars_merged(
 #'   vs,
 #'   dataset_add = select(dm, -DOMAIN),
 #'   by_vars = exprs(STUDYID, USUBJID)
 #' ) %>%
-#'   select(STUDYID, USUBJID, VSTESTCD, VISIT, VSSTRESN, AGE, AGEU)
-#'
+#'   select(USUBJID, VSTESTCD, VISIT, VSSTRESN, AGE, AGEU)
+#'
+#' @caption Merge only the first/last value (`order` and `mode`)
+#' @info Merge the last occurring weight for each subject to the demographics dataset.
+#'
+#' - To enable sorting by visit date `convert_dtc_to_dtm()` is used to convert
+#'   to a datetime, within the `order` argument.
+#' - Then the `mode` argument is set to `"last"` to ensure the last sorted value
+#'   is taken. Be cautious if `NA` values are possible in the `order` variables -
+#'   see [Sort Order](https://pharmaverse.github.io/admiral/articles/generic.html#sort_order).
```

Comment:
Could add mention of what happens to patient 03 here (see my comment above)

Reply:
See above reply